presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+
+        // If the table is external, we won't clean up the Accumulo tables
+        if (table.isExternal()) {
+            return;
+        }
+
+        try {
+            // delete the table
+            conn.tableOperations().delete(tn);
+
+            // and delete our index tables
+            if (table.isIndexed()) {
+                conn.tableOperations().delete(Indexer.getIndexTableName(stName));
+                conn.tableOperations().delete(Indexer.getMetricsTableName(stName));
+            }
+        }
+        catch (Exception e) {
+            throw new PrestoException(INTERNAL_ERROR, "Failed to delete table",
+                    e);
+        }
+    }
+
+    /**
+     * Renames the metadata associated Accumulo table
+     *
+     * @param oldName Old table name
+     * @param newName New table name
+     */
+    public void renameTable(SchemaTableName oldName, SchemaTableName newName)
+    {
+        if (this.getTable(newName) != null) {
+            throw new PrestoException(ALREADY_EXISTS,
+                    "Table " + newName + " already exists");
+        }
+
+        if (!oldName.getSchemaName().equals(newName.getSchemaName())) {
+            throw new PrestoException(NOT_SUPPORTED,
+                    "Accumulo does not support renaming tables to different namespaces (schemas)");
+        }
+
+        AccumuloTable oldTable = getTable(oldName);
+        AccumuloTable newTable = new AccumuloTable(newName.getSchemaName(), newName.getTableName(),
+                oldTable.getColumns(), oldTable.getRowId(), oldTable.isExternal(),
+                oldTable.getSerializerClassName(), oldTable.getScanAuthorizations());
+
+        // Probably being over-cautious on the failure conditions and rollbacks, but I suppose it is
+        // better to be safe than sink ships.
+
+        try {
+            // First, rename the Accumulo table in the event there is some issue
+            conn.tableOperations().rename(oldTable.getFullTableName(), newTable.getFullTableName());
+        }
+        catch (AccumuloSecurityException | TableNotFoundException | AccumuloException
+                | TableExistsException e) {
+            throw new PrestoException(INTERNAL_ERROR, "Failed to rename table",
+                    e);
+        }
+
+        // Rename index tables as well, if indexed, of course
+        if (oldTable.isIndexed()) {
+            try {
+                conn.tableOperations().rename(oldTable.getIndexTableName(),
+                        newTable.getIndexTableName());
+            }
+            catch (AccumuloSecurityException | TableNotFoundException | AccumuloException
+                    | TableExistsException e) {
+                try {
+                    conn.tableOperations().rename(newTable.getFullTableName(),
+                            oldTable.getFullTableName());
+                }
+                catch (Exception e1) {
+                    throw new PrestoException(INTERNAL_ERROR,
```

Comment:
Going to validate rename operation first, then execute.  Will attempt to rollback in transaction and point user to documentation on error.